### PR TITLE
Feat(medusa): update migration and initDb to run link-migrations

### PR DESCRIPTION
**Pull request by pKorsholm**

### CODE REVIEW

File: packages/medusa/src/loaders/medusa-app.ts
Comment (olivermrbl) [inline packages/medusa/src/loaders/medusa-app.ts]: praise(this-file): nice

File: packages/medusa/src/loaders/medusa-app.ts
Comment (olivermrbl) [inline packages/medusa/src/loaders/medusa-app.ts]: suggestion: `MedusaContainer` type

Comment (olivermrbl) [reply packages/medusa/src/loaders/medusa-app.ts]: Perhaps extract all loader args to a type

File: packages/medusa/src/loaders/medusa-app.ts
Comment (olivermrbl) [inline packages/medusa/src/loaders/medusa-app.ts]: question: we moved from having the pg connection in injected deps to now passing the clientUrl in shared resources – curious to know why?

Also, do we need a todo or something here for the empty object?

File: packages/medusa/src/loaders/medusa-app.ts
Comment (olivermrbl) [inline packages/medusa/src/loaders/medusa-app.ts]: suggestion(non-blocking): -> `registerInContainer`

File: integration-tests/plugins/__tests__/pricing/get-product.ts
Comment (olivermrbl) [inline integration-tests/plugins/__tests__/pricing/get-product.ts]: todo: rm
```suggestion
```

**Review by olivermrbl — COMMENTED** [review]
@adrien2p @carlos-r-l-rodrigues would love if one of you could have a look at this one too 🙏

File: packages/modules-sdk/src/remote-query.ts
Comment (adrien2p) [inline packages/modules-sdk/src/remote-query.ts]: todo: I ll have a review tomorrow morning unless @carlos-r-l-rodrigues review it before me 😅 But this section should be removed. I don't think It is the appropriate fix but more of a work around for a bug which is that here the reference of serviceConfig is mutated and therefore impact the number of elements in it later on. Especially when having multiple medusaApp with the tests. We fixed it in our pr and in reality we should use a copy of serviceConfig here before pushing. I would suggest to do servicesConfig_ = [...servicesConfig]

File: packages/medusa/src/loaders/medusa-app.ts
Comment (adrien2p) [reply packages/medusa/src/loaders/medusa-app.ts]: It should be the connection, otherwise it is not shared anymore

File: packages/medusa/src/loaders/medusa-app.ts
Comment (adrien2p) [inline packages/medusa/src/loaders/medusa-app.ts]: I wouldn't register all of that in the container. But instead return the return from the index loader such as we do for the container. Than you can use it just like we do for the container when calling the bootstrap. Wdyt?

File: packages/medusa/src/loaders/medusa-app.ts
Comment (adrien2p) [reply packages/medusa/src/loaders/medusa-app.ts]: Praise too, will have lot of conflicts as it will change quite a lot with out pr which is something we will probably extract to a separate pr 😅 as there is some issues.

Cc @carlos-r-l-rodrigues

**Review by adrien2p — CHANGES_REQUESTED** [review]
Looks pretty good, few things to handle but otherwise 👍

**Review by olivermrbl — APPROVED** [review]
LGTM, pending @adrien2p's approval

**Review by carlos-r-l-rodrigues — APPROVED** [review]
Great Job... thank you

File: packages/medusa/src/loaders/medusa-app.ts
Comment (pKorsholm) [reply packages/medusa/src/loaders/medusa-app.ts]: we need at least the link modules when we get to the workflow handlers, how would you prefer handling this, registering just the link service?

File: integration-tests/environment-helpers/use-db.js
Comment (adrien2p) [inline integration-tests/environment-helpers/use-db.js]: question: why don't we pass the config module directly?

File: integration-tests/environment-helpers/use-db.js
Comment (adrien2p) [inline integration-tests/environment-helpers/use-db.js]: todo: rm log

File: integration-tests/environment-helpers/use-db.js
Comment (adrien2p) [inline integration-tests/environment-helpers/use-db.js]: todo: since this is only used under some conditions, should we require it in the if block along side the rest?

File: integration-tests/environment-helpers/use-db.js
Comment (adrien2p) [inline integration-tests/environment-helpers/use-db.js]: suggestion: we could move that closer to where they are used wdyt?

File: integration-tests/environment-helpers/use-db.js
Comment (adrien2p) [inline integration-tests/environment-helpers/use-db.js:155]: question: if one flag is enabled, will it run the migration for all links?

File: integration-tests/plugins/__tests__/pricing/get-product.ts
Comment (adrien2p) [inline integration-tests/plugins/__tests__/pricing/get-product.ts]: suggestion: `ModuleRegistrationName.PRICING`?

File: integration-tests/plugins/__tests__/pricing/get-product.ts
Comment (adrien2p) [inline integration-tests/plugins/__tests__/pricing/get-product.ts]: todo: as we discussed, we should probably just register the link like we do for the modules, but not the entire medusa app

File: integration-tests/plugins/__tests__/pricing/get-product.ts
Comment (adrien2p) [inline integration-tests/plugins/__tests__/pricing/get-product.ts]: todo: I think there is some missing words in the description no?

File: packages/medusa/src/loaders/medusa-app.ts
Comment (adrien2p) [inline packages/medusa/src/loaders/medusa-app.ts:32]: todo: I think that we might be missing the extra as well in case there is some ssl stuff wdyt? havn't tested

File: packages/medusa/src/loaders/medusa-app.ts
Comment (adrien2p) [reply packages/medusa/src/loaders/medusa-app.ts]: yes, registering just the links like we do for the modules

File: integration-tests/environment-helpers/use-db.js
Comment (adrien2p) [inline integration-tests/environment-helpers/use-db.js]: todo: `registerInContainer`

File: packages/medusa/src/scripts/money-amount-pricing-module-migration.ts
Comment (adrien2p) [inline packages/medusa/src/scripts/money-amount-pricing-module-migration.ts]: todo: since someone could try to run the script without the correct configuration, I would think that here we should check that link has a value right? also to be sure that if something goes wrong in the link modules we can warn the user instead of cannot read create from undefined. wdyt?

File: packages/modules-sdk/src/remote-query.ts
Comment (adrien2p) [inline packages/modules-sdk/src/remote-query.ts:32]: praise 🙏

File: packages/workflows/src/handlers/product/update-products-variants-prices.ts
Comment (adrien2p) [inline packages/workflows/src/handlers/product/update-products-variants-prices.ts]: todo: change to resolve the link instead

**Review by adrien2p — CHANGES_REQUESTED** [review]
Graet job man, few todo's and some comments to look at and then it should be good to go

File: integration-tests/environment-helpers/use-db.js
Comment (pKorsholm) [reply integration-tests/environment-helpers/use-db.js:155]: no, `runMigrations` validates each link to check both `primary` and `secondary` services are among the loaded module keys

**Review by adrien2p — COMMENTED** [review]
LGTM, just need to resolve conflicts

**Review by adrien2p — APPROVED** [review]

**Pull request merged**
